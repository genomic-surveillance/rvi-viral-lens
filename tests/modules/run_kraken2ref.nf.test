nextflow_process {

    name "Test Process run_kraken2ref process"
    script "modules/run_kraken2ref.nf"
    process "run_kraken2ref"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                meta = [
                    id:"test.9000000",
                    sample_id:"test",
                    taxid:9000000
                ]
                containers_dir="/lustre/scratch126/gsu/team112/personal/ad45/rvi_box/dev_containers/"
                depth_treshold=5
                test_loc="tests/test_data/"
            }

            process {

            """
            // define inputs of the process here. Example:
            input[0] = tuple(
                params.meta, 
                file("${params.test_loc}/test_kraken_outputs/test.kraken.output"),
                [
                    file("${params.test_loc}/test_kraken_outputs/test.class_seqs_1.fq"),
                    file("${params.test_loc}/test_kraken_outputs/test.class_seqs_2.fq")
                ],
                file("${params.test_loc}/test_kraken_outputs/test.report.txt"))
            """

            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}
