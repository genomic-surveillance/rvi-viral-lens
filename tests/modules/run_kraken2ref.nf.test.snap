{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test.9000000",
                            "taxid": 9000000,
                            "sample_id": "test"
                        },
                        [
                            "test.9000000_2697049_R1.fq:md5,b0eaac69db411379704837a2b2a9dbd8",
                            "test.9000000_2697049_R2.fq:md5,705ce6f753733d8ba3ccdffad193ef52",
                            "test.9000000_9000016_R1.fq:md5,956b247146d66af80f328a791cacd1e9",
                            "test.9000000_9000016_R2.fq:md5,4d4b144a3f6947a5a9f6603fbd0dd4d1",
                            "test.9000000_9000017_R1.fq:md5,956b247146d66af80f328a791cacd1e9",
                            "test.9000000_9000017_R2.fq:md5,4d4b144a3f6947a5a9f6603fbd0dd4d1",
                            "test.9000000_9000018_R1.fq:md5,956b247146d66af80f328a791cacd1e9",
                            "test.9000000_9000018_R2.fq:md5,4d4b144a3f6947a5a9f6603fbd0dd4d1",
                            "test.9000000_9000019_R1.fq:md5,956b247146d66af80f328a791cacd1e9",
                            "test.9000000_9000019_R2.fq:md5,4d4b144a3f6947a5a9f6603fbd0dd4d1",
                            "test.9000000_9000020_R1.fq:md5,956b247146d66af80f328a791cacd1e9",
                            "test.9000000_9000020_R2.fq:md5,4d4b144a3f6947a5a9f6603fbd0dd4d1",
                            "test.9000000_9000022_R1.fq:md5,956b247146d66af80f328a791cacd1e9",
                            "test.9000000_9000022_R2.fq:md5,4d4b144a3f6947a5a9f6603fbd0dd4d1"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2024-03-05T20:56:05+0000"
    }
}